lib/hello/catalog.ex

```diff
@@ -0,0 +1,229 @@
+defmodule Hello.Catalog do
+  @moduledoc """
+  The Catalog context.
+  """
+
+  import Ecto.Query, warn: false
+  alias Hello.Repo
+
+  alias Hello.Catalog.Product
+  alias Hello.Catalog.Category
+
+  @doc """
+  Returns the list of products.
+
+  ## Examples
+
+      iex> list_products()
+      [%Product{}, ...]
+
+  """
+  def list_products do
+    Repo.all(Product)
+  end
+
+  @doc """
+  Gets a single product.
+
+  Raises `Ecto.NoResultsError` if the Product does not exist.
+
+  ## Examples
+
+      iex> get_product!(123)
+      %Product{}
+
+      iex> get_product!(456)
+      ** (Ecto.NoResultsError)
+
+  """
+  #def get_product!(id), do: Repo.get!(Product, id)
+  # We load the categories for a specific product
+  def get_product!(id) do
+    Product
+    |> Repo.get(id)
+    |> Repo.preload(:categories)
```

Comment:
@keliumJU what does a preload do?

Reply:
In this case, it preloads a list of categories based on the id of the product, that is, it returns a list of the categories that a product has, this is possible because there is a categories table